ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -237,14 +244,18 @@ public void run() {
     }
   }
 
-  private void scheduleCompactionIfRequired(CompactionInfo ci, Table t, Partition p, String runAs, boolean metricsEnabled)
+  private void scheduleCompactionIfRequired(CompactionInfo ci, Database d, Table t, Partition p, String runAs, boolean metricsEnabled)
       throws MetaException {
     StorageDescriptor sd = resolveStorageDescriptor(t, p);
     try {
       ValidWriteIdList validWriteIds = resolveValidWriteIds(t);
       CompactionType type = checkForCompaction(ci, validWriteIds, sd, t.getParameters(), runAs);
       if (type != null) {
         ci.type = type;
+        ci.poolName = t.getParameters().get(Constants.HIVE_COMPACTOR_WORKER_POOL);
```

Comment:
could we extract this logic into a separate method, like getPoolName, and pass it as param to scheduleCompactionIfRequired?